### ISSUE TITLE
Remove disambiguator from intra doc link text

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust#65354]
```rust
/// Trait: [`trait@Name`], fn: [`fn@Name`]
pub struct S;

pub trait Name {}

#[allow(non_snake_case)]
pub fn Name() {}
```

This is currently rendered as:

> <img src="https://user-images.githubusercontent.com/1940490/66709897-4a232900-ed22-11e9-9069-0ed584e86df5.png" width="50%">

The links point to the right place (the trait and the fn respectively) but I believe the link text should be just "`Name`" for both.

Discovered in https://github.com/rust-lang/rust/pull/65336.

*Mentioning the tracking issue: #43466*

This issue has been assigned to @jyn514 via [this comment](https://github.com/rust-lang/rust/issues/65354#issuecomment-653978812).

**Comment by GuillaumeGomez** [comment]
I can do it if you want but maybe you prefer doing it?

**Comment by dtolnay** [comment]
I definitely don't prefer doing it. :wink: Feel free to go ahead. Thanks!

**Comment by GuillaumeGomez** [comment]
Ok haha. Will try to do it not too far into the future.

**Comment by jyn514** [comment]
So, I tried implementing this just now and it's not possible with the current pulldown-cmark API. The issue is that while `broken_link_callback` can return a new title (for when you hover) and a new URL, it cannot change the text that is rendered for shortcut reference links: [`fn@Name`]. See https://github.com/raphlinus/pulldown-cmark/blob/master/src/parse.rs#L2213.

Maybe we could ask pulldown to extend the API to allow this? I had trouble with the last PR I made though, the link parsing is very complicated.
